backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +72,98 @@ def disambiguate_signatures(self, clusters):
     db.session.commit()
 
 
+def match_literature_author(author_data, matcher_config):
+    matched_records = (
+        matched_record["inner_hits"]["authors"]["hits"]["hits"]
+        for matched_record in match(author_data, matcher_config)
+    )
+    matched_authors_refs = []
+    for record_authors in matched_records:
+        matched_authors_refs.extend(
+            author["_source"]["record"]["$ref"]
+            for author in record_authors
+            if "record" in author["_source"]
+        )
+    if len(set(matched_authors_refs)) == 1:
+        author_reference_url = matched_authors_refs[0]
+        return {"$ref": author_reference_url}
+
+
+def assign_reference_to_author(author, updated_authors, reference):
+    author["record"] = reference
+    updated_authors.append(reference["$ref"].split("/")[-1])
+
+
+def match_literature_author_with_configs(author, updated_authors, record):
+    configs = [
+        "AUTHOR_MATCHER_NAME_CONFIG",
+        "AUTHOR_MATCHER_NAME_INITIALS_CONFIG",
+        "AUTHOR_MATCHER_COLLABORATION_CONFIG",
+        "AUTHOR_MATCHER_AFFILIATION_CONFIG",
+    ]
+
+    parsed_name = ParsedName.loads(author.get("full_name"))
+    author_matcher_data = {
+        "first_name": parsed_name.first,
+        "last_name": parsed_name.last,
+        "full_name": author.get("full_name"),
+        "collaborations": get_value(record, "collaborations.value", []),
+        "affiliations": get_value(author, "affiliations.value", []),
+    }
+
+    for config in configs:
+        matched_recid = match_literature_author(
+            author_matcher_data, current_app.config[config]
+        )
+        if matched_recid:
+            assign_reference_to_author(author, updated_authors, matched_recid)
+            return author
+
+
+def create_new_author(author, record):
```

Comment:
Instead of creating yet another create author function, we could refactor https://github.com/inspirehep/inspirehep/blob/50a36d3029a96ce9a41852486c1d81a9b6e7c9e3/backend/inspirehep/disambiguation/utils.py#L75 to take `data` param and rename it to `create_author`.